sdk/search/azure-search-documents/azure/search/documents/_internal/aio/_search_client_async.py

```diff
@@ -139,22 +137,50 @@ async def search(self, query, **kwargs):
                 :dedent: 4
                 :caption: Get search result facets.
         """
-        if isinstance(query, six.string_types):
-            query = SearchQuery(search_text=query)
-        elif not isinstance(query, SearchQuery):
-            raise TypeError(
-                "Expected a string or SearchQuery for 'query', but got {}".format(
-                    repr(query)
-                )
-            )
+        include_total_result_count = kwargs.pop("include_total_result_count", None)
+        facets = kwargs.pop("facets", None)
+        filter_arg = kwargs.pop("filter", None)
+        highlight_fields = kwargs.pop("highlight_fields", None)
+        highlight_post_tag = kwargs.pop("highlight_post_tag", None)
+        highlight_pre_tag = kwargs.pop("highlight_pre_tag", None)
+        minimum_coverage = kwargs.pop("minimum_coverage", None)
+        order_by = kwargs.pop("order_by", None)
+        query_type = kwargs.pop("query_type", None)
+        scoring_parameters = kwargs.pop("scoring_parameters", None)
+        scoring_profile = kwargs.pop("scoring_profile", None)
+        search_fields = kwargs.pop("search_fields", None)
+        search_mode = kwargs.pop("search_mode", None)
+        select = kwargs.pop("select", None)
+        skip = kwargs.pop("skip", None)
+        top = kwargs.pop("top", None)
+        query = SearchQuery(
+            search_text=search_text,
+            include_total_result_count=include_total_result_count,
+            facets=facets,
+            filter=filter_arg,
+            highlight_fields=highlight_fields,
+            highlight_post_tag=highlight_post_tag,
+            highlight_pre_tag=highlight_pre_tag,
+            minimum_coverage=minimum_coverage,
+            order_by=order_by,
+            query_type=query_type,
+            scoring_parameters=scoring_parameters,
+            scoring_profile=scoring_profile,
+            search_fields=search_fields,
+            search_mode=search_mode,
+            select=select,
+            skip=skip,
+            top=top
+        )
```

Comment:
Nit - Is it worth moving this logic onto the SearchQuery class so you don't have to update both the sync/async clients whenever we add a new parameter?  I'm not sure how the Python folks typically handle these situations.  (Also, we can just file a work item rather for Preview 5 rather than try to fix it today.)

Reply:
SearchQuery and AutocompleteQuery, SuggestQuery share one ctor. If we put all of pop operation there, I am afraid it will be messed up.